### ISSUE TITLE
Loss function can not converge using the AdamOptimizer

### ISSUE THREAD

**Issue by MuZiGuYue** [kratzert/finetune_alexnet_with_tensorflow#68]
Hi! It's very kind of you for sharing the repository. I have two questions about this code. 
1. After changing GradientDescentOptimizer to AdamOptimizer, the loss function cannot converge.
2. I have some questions when I read other people’s questions. 
`train_layers = ['fc8', 'fc7', 'fc6']`  indicates the layers we don't want to train.
`var = tf.get_variable('biases', trainable=False)`  indicates that the training parameters of these layers are unchanged, and the corresponding parameters in bvlc_alexnet.npy are maintained.
That is to say, the parameters in ['fc8', 'fc7', 'fc6'] remain unchanged during training, and only the parameters of other layers are changed. So how do you initialize the trainable parameters in other layers? Do they use the corresponding parameters in bvlc_alexnet.npy or the default random initialization?

**Comment by kratzert** [comment]
1. Never tried that out and should not happen. Try adapting the learning rate. It should converge same as SGD.
2. Think again: the name 'train_layers' of course does indicate the layers you DO want to train. The layers in this list are not loaded from the pre-trained weights but randomly initialized from scratch

**Comment by Novicemonk** [comment]
I've successfully used the AdamOptimizer in tensorflow with this code. It required setting a very low learning rate (~0.001) & using the other default parameters for this optimizer. It took about 17 epochs to fully converge on my dataset.
